test/dynamo/test_misc.py

```diff
@@ -229,6 +229,61 @@ def boolarg(aa, bb, flag):
         self.assertTrue(same(val4, correct1))
         self.assertEqual(counter.frame_count, 3)
 
+    def test_torch_dispatch_ignore_compile_internals(self):
+        counters.clear()
+        from torch.utils._python_dispatch import TorchDispatchMode
+
+        @torch.library.custom_op("mylib::foo", mutates_args=())
+        def foo(x: torch.Tensor) -> torch.Tensor:
+            return x.clone()
+
+        def checksum(x):
+            return x.abs().sum()
+
+        _checksums = []
+
+        class ChecksumFoo(TorchDispatchMode):
+            @classmethod
+            def ignore_compile_internals(cls):
+                return True
+
+            def __init__(self) -> None:
+                super().__init__()
+
+            def __torch_dispatch__(self, func, types, args, kwargs=None):
+                kwargs = kwargs or {}
+
+                if func is torch.ops.mylib.foo.default:
+                    # Do some compute, smoketest to see if there's a bad interaction
+                    _checksums.append(args[0].abs().sum())
+
+                return func(*args, **kwargs)
+
+        backend_count = 0
+
+        def backend(gm, args):
+            global backend_count
+            backend_count += 1
+            return gm
+
+        # test e2e, with Inductor, as smoketest.
+        @torch.compile(fullgraph=True, backend="inductor")
+        def g(x):
+            return 2 * x.sin().cos()
+
+        x = torch.randn(3)
+
+        with ChecksumFoo():
+            foo(x)
+            g(x)
+            foo(x)
+
+        self.assertEqual(len(_checksums), 2)
+        # If you are getting "3" here, then that means the .abs().sum()
```

Comment:
hmm i'm a bit confused by the test - even without your change, wouldnt' the `.abs().sum()` never get compiled? we only call that in the case when our TorchDispatchMode intercepts the custom op foo, but it doesn't look like the test ever calls `mylib.foo` in a compiled region.

Reply:
I'm just going to delete the comment because I don't know what exactly is going on here. The .abs().sum() isn't being compiled. When Dynamo falls back to eager, it turns out that dynamo decides to compile each of the ops individually (the mul, sin, cos). This means that it had an opportunity to turn itself back on, but I don't know why or how.

Reply:
This was a bug in my implementation (great catch!)